src/route-recognizer.js

```diff
@@ -37,6 +48,7 @@ export class RouteRecognizer {
     let routeName = route.handler.name;
     let isEmpty = true;
     let segments = parse(route.path, names, types);
+    const caseSensitiveRoute = route.caseSensitive === undefined ? this.caseSensitive : route.caseSensitive;
```

Comment:
I think it'd be better to check if `route.caseSensitive` is a boolean, to guard against invalid input.